doc/api/fs.md

```diff
@@ -13,1160 +13,1284 @@ way modeled on standard POSIX functions.
 
 To use this module:
 
-```js
-const fs = require('fs');
+```js esm
+// Using ESM Module syntax:
+import * as fs from 'fs';  // Sync and callback APIs
 ```
 
-All file system operations have synchronous, callback, and promise-based
-forms.
-
-## Synchronous example
-
-The synchronous form blocks the Node.js event loop and further JavaScript
-execution until the operation is complete. Exceptions are thrown immediately
-and can be handled using `try…catch`, or can be allowed to bubble up.
+```js esm
+import * as fs from 'fs/promises';  // Promise APIs
+```
 
-```js
-const fs = require('fs');
+```js cjs
+// Using CommonJS syntax:
+const fs = require('fs');  // Sync and callback APIs
+```
 
-try {
-  fs.unlinkSync('/tmp/hello');
-  console.log('successfully deleted /tmp/hello');
-} catch (err) {
-  // handle the error
-}
+```js cjs
+const fs = require('fs/promises');  // Promise APIs
 ```
 
-## Callback example
+All file system operations have synchronous, callback, and promise-based
+forms, and are accessible using both CommonJS syntax and ES6 Modules (ESM).
 
-The callback form takes a completion callback function as its last
-argument and invokes the operation asynchronously. The arguments passed to
-the completion callback depend on the method, but the first argument is always
-reserved for an exception. If the operation is completed successfully, then
-the first argument is `null` or `undefined`.
+## Promise example
 
-```js
-const fs = require('fs');
+Promise-based operations return a promise that is resolved when the
+asynchronous operation is complete.
 
-fs.unlink('/tmp/hello', (err) => {
-  if (err) throw err;
+```js esm
+// Using ESM Module syntax:
+import { unlink } from 'fs/promises';
+
+try {
+  await unlink('/tmp/hello');
   console.log('successfully deleted /tmp/hello');
-});
+} catch (error) {
+  console.error('there was an error:', error.message);
+}
 ```
 
-## Promise example
-
-Promise-based operations return a `Promise` that is resolved when the
-asynchronous operation is complete.
-
-```js
-const fs = require('fs/promises');
+```js cjs
+// Using CommonJS syntax
```

Comment:
```suggestion
// Using CommonJS syntax
'use strict';
```

Reply:
I'd prefer to omit the `use strict` from the examples